pkg/cmd/render/render.go

```diff
@@ -0,0 +1,196 @@
+	performancev2 "github.com/openshift-kni/performance-addon-operators/api/v2"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components/kubeletconfig"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components/machineconfig"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components/runtimeclass"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components/tuned"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+type renderOpts struct {
+	ppInputFileDir string
+	assetsInDir    string
+	assetsOutDir   string
+}
+
+type manifestable map[string]v1.Object
+
+//NewRenderCommand creates a render command.
+func NewRenderCommand() *cobra.Command {
+	renderOpts := renderOpts{}
+
+	cmd := &cobra.Command{
+		Use:   "render",
+		Short: "Render performance-addon-operator manifests",
+		Long: `Render performance-addon-operator manifests creates the following manifests:
+		       - KubeletConfig.yaml
+			   - MachineConfig.yaml
+			   - RuntimeClass.yaml
+			   - Tuned.yaml`,
+		Run: func(cmd *cobra.Command, args []string) {
+
+			if err := renderOpts.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+
+			if err := renderOpts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	renderOpts.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (r *renderOpts) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&r.ppInputFileDir, "performance-profile-dir-path", r.ppInputFileDir, "Input path for the Performance Profile manifest's directory.")
+	fs.StringVar(&r.assetsInDir, "asset-input-dir", components.AssetsDir, "Input path for the assets directory.")
+	fs.StringVar(&r.assetsOutDir, "asset-output-dir", r.assetsOutDir, "Output path for the rendered manifests.")
+	// environmet variables has precedence over standard input
+	r.readFlagsFromEnv()
+}
+
+func (r *renderOpts) readFlagsFromEnv() {
+	if ppDir := os.Getenv("PP_DIR"); len(ppDir) > 0 {
+		r.ppInputFileDir = ppDir
+	}
+
+	if assetInDir := os.Getenv("ASSET_IN_DIR"); len(assetInDir) > 0 {
+		r.assetsInDir = assetInDir
+	}
+
+	if assetsOutDir := os.Getenv("ASSET_OUT_DIR"); len(assetsOutDir) > 0 {
+		r.assetsOutDir = assetsOutDir
+	}
+}
+
+func (r *renderOpts) Validate() error {
+	if len(r.ppInputFileDir) == 0 {
+		return fmt.Errorf("performance-profile-file-path must be specified")
+	}
+
+	if len(r.assetsOutDir) == 0 {
+		return fmt.Errorf("asset-output-dir must be specified")
+	}
+
+	return nil
+}
+
+func (r *renderOpts) Run() error {
+	PPFiles := getPPList(r.ppInputFileDir)
+	if len(PPFiles) == 0 {
+		return fmt.Errorf("no valid performance-profiles found under: %v", r.ppInputFileDir)
+	}
+
+	for idx, pp := range PPFiles {
```

Comment:
this looks similar but different wrt to `applyComponents`. We should try harder NOT to have two similar but distinct flows. Can we factor out the two flows in a common path?

Reply:
I thought we agreed to dedicate a separate PR for this, but I can work on that np.

Reply:
I don't recall, it could very much be :) but if so that PR should come right after this one, we should not carry this tech debt for too long.